main.nf

```diff
@@ -78,24 +79,21 @@ process BuildDockerContainers {
 dockerContainersBuilt = dockerContainersBuilt.view {"Docker container: $repository/$it:$tag built."}
 
 process BuildSingularityContainers {
-  tag {container}
+  tag {repository + "/" + container + ":" + tag}
+
+  publishDir publishSingularityDir, mode: 'copy'
 
   input:
     val container from singularityContainers
 
   output:
-    val container into singularityContainersBuilt
+    file("*.img") into singularityContainersBuilt
 
   when: singularity
 
   script:
   """
-  docker run \
-  -v /var/run/docker.sock:/var/run/docker.sock \
-  -v /tmp/$container:/output \
-  --privileged -t --rm \
-  singularityware/docker2singularity \
-  $repository/$container:$tag
+  singularity pull --name $container-${tag}.img docker://$repository/$container:$tag
```

Comment:
@ewels
I'm just using this command line to download the right docker container, and giving it a more convenient name.
(If you do have a singularity-hub container, you can use `shub://$repository/$container` instead of the `docker...`)
So that it correspond already to the path specified in the `singularity.config` file from CAW.
This `singularity.config` is a modified version of our `docker.config` that just countains paths to the singularity image instead of the names of the docker containers.
And that was all, it's that simple...